BENCHMARK.mb

```diff
@@ -1,239 +1,253 @@
-BM_DeltaDecode/64         15.0 ns         15.0 ns     46240484 bytes_per_second=15.9266Gi/s
-BM_DeltaDecode/512         119 ns          119 ns      5847987 bytes_per_second=16.0084Gi/s
-BM_DeltaDecode/4096        986 ns          986 ns       718277 bytes_per_second=15.4807Gi/s
-BM_DeltaDecode/8192       2249 ns         2249 ns       310347 bytes_per_second=13.5674Gi/s
+-----------------------------------------------------------------------------------
+Benchmark                         Time             CPU   Iterations UserCounters...
+-----------------------------------------------------------------------------------
+BM_DeltaEncode/8               2.77 ns         2.77 ns    255552797 bytes_per_second=10.7765Gi/s
+BM_DeltaEncode/64              20.0 ns         20.0 ns     34724324 bytes_per_second=11.9311Gi/s
+BM_DeltaEncode/512              157 ns          157 ns      4477020 bytes_per_second=12.1179Gi/s
+BM_DeltaEncode/4096            1349 ns         1349 ns       511078 bytes_per_second=11.3131Gi/s
+BM_DeltaEncode/32768          10248 ns        10249 ns        66830 bytes_per_second=11.9108Gi/s
+BM_DeltaEncode/65536          21037 ns        21039 ns        32294 bytes_per_second=11.6042Gi/s
+BM_DeltaEncodeSIMD/8           3.16 ns         3.16 ns    219127790 bytes_per_second=9.44171Gi/s
+BM_DeltaEncodeSIMD/64          14.3 ns         14.3 ns     48618910 bytes_per_second=16.6623Gi/s
+BM_DeltaEncodeSIMD/512         96.4 ns         96.4 ns      7333191 bytes_per_second=19.7942Gi/s
+BM_DeltaEncodeSIMD/4096         841 ns          841 ns       812650 bytes_per_second=18.1341Gi/s
+BM_DeltaEncodeSIMD/32768       6960 ns         6961 ns       100257 bytes_per_second=17.5361Gi/s
+BM_DeltaEncodeSIMD/65536      13886 ns        13887 ns        49170 bytes_per_second=17.5805Gi/s
+BM_DeltaDecode/8               2.24 ns         2.24 ns    311170303 bytes_per_second=13.3241Gi/s
+BM_DeltaDecode/64              15.3 ns         15.3 ns     45601031 bytes_per_second=15.624Gi/s
+BM_DeltaDecode/512              118 ns          118 ns      5795450 bytes_per_second=16.2234Gi/s
+BM_DeltaDecode/4096             958 ns          958 ns       745372 bytes_per_second=15.922Gi/s
+BM_DeltaDecode/32768           9547 ns         9548 ns        81424 bytes_per_second=12.7852Gi/s
+BM_DeltaDecode/65536          18500 ns        18501 ns        36067 bytes_per_second=13.1959Gi/s
+BM_DeltaDecodeSIMD/8           2.54 ns         2.54 ns    275182544 bytes_per_second=11.749Gi/s
+BM_DeltaDecodeSIMD/64          18.9 ns         18.9 ns     36562479 bytes_per_second=12.6207Gi/s
+BM_DeltaDecodeSIMD/512          227 ns          227 ns      3128738 bytes_per_second=8.39837Gi/s
+BM_DeltaDecodeSIMD/4096        2035 ns         2035 ns       339673 bytes_per_second=7.49841Gi/s
+BM_DeltaDecodeSIMD/32768      16525 ns        16526 ns        42024 bytes_per_second=7.38653Gi/s
+BM_DeltaDecodeSIMD/65536      33179 ns        33182 ns        20914 bytes_per_second=7.35773Gi/s
```

Comment:
DeltaDecodeSIMD take 33182ns when normal one take only 18501ns. I will try to upgrade this.
@gregulator Please check is there anything wrong with my implementation.

Reply:
improve performance with last commit, bellow you can see current result
BM_DeltaDecode/65536              18394 ns        18391 ns        37994 bytes_per_second=13.2747Gi/s
BM_DeltaDecodeSIMD/65536      15078 ns        15078 ns        46292 bytes_per_second=16.1919Gi/s